src/scheduler/worker.rs

```diff
@@ -31,6 +31,11 @@ pub fn current_worker_ordinal() -> Option<ThreadId> {
 pub struct GCWorkerShared<VM: VMBinding> {
     /// Worker-local statistics data.
     stat: AtomicRefCell<WorkerLocalStat<VM>>,
+    /// Accumulated bytes for live objects in this GC. When each worker scans
+    /// objects, we increase the live bytes. We get this value from each worker
+    /// at the end of a GC, and reset this counter.
+    #[cfg(feature = "count_live_bytes_in_gc")]
+    live_bytes: AtomicUsize,
```

Comment:
Do we need atomic for this? It's worker-local right?

Reply:
Ah OK. I was just thinking if we can make it cheap enough by not using atomic for worker local stats, we could enable this feature by default.

Reply:
Why not make it worker-local and each worker reports the live bytes after their transitive closure? Then we don't have to explicitly iterate through all the workers, we just collate what the numbers we get from the transitive closure.

Reply:
It could be any stage after the transitive closure stages (including weak reference processing stages).  `ReleaseMutator` should be an ideal place to collect worker-local data.

Reply:
We have a `ReleaseCollector` work packet.  It currently releases only the copy context.  We can also use this stage to add worker-local bytes to the global `AtomicUsize`, and clear worker-local data.

Reply:
We need to schedule a 'designated' work packet for each worker (which makes sure the work packet is executed for every worker, and only once for each worker). @wks I think you meant `ReleaseCollector`, not `ReleaseMutator` (which is not 'designated work').

But what should we do for mark compact? For mark compact, we need to either reset the live bytes after the first transitive closure, or not count live bytes in the fist transitive closure. Currently, I can easily iterate through all the workers, and clear their counters. With this change, we would need another designated work packet to reset the counter for mark compact between the two transitive closures.

But anyway, I don't feel we want to enable this by default. 
1. This is only intended as a debug feature to know about fragmentation. Unlike GCs in some languages, we do not count allocated bytes during allocation. This can hardly be a useful stat even if we turn it on by default.
2. we need to call to the binding to get object size, which may not be cheap for some bindings. Enabling this by default does not sound like a good idea.

Let me know what you think.

Reply:
> Maybe it's easier to defer the "live byte count" to each policy as opposed to each worker? But then it stops being worker-local so it would need synchronization.

Yeah. It need at least the same level synchronization as the current code. And the counting code is scattered to each policy.

Reply:
We could use `AtomicUsize::as_mut_ptr` and do non-atomic add with unsafe code when we do the counting. But I still question if we want to make this enabled by default.

Reply:
I personally don't have a strong opinion. Keeping track of fragmentation is definitely useful though. If it's possible to do it cheaply for every GC, may as well do it. If it's not possible then I don't think should spend more time on it.

Reply:
@qinsoon Sorry about the typo. I meant `ReleaseCollector`.  Yes.  It will be a problem for `MarkCompact`.  If we make the counter a private field of a `Worker`, it will require another rendezvous (designated works) between the marking phase and the forwarding phase.  Given this situation, I don't mind if we use `AtomicUsize` for now.